python/cudf/cudf/tests/test_string.py

```diff
@@ -2214,7 +2218,11 @@ def test_string_str_rindex(data, sub, er):
 
     if er is None:
         assert_eq(ps.str.rindex(sub), gs.str.rindex(sub), check_dtype=False)
-        assert_eq(pd.Index(ps).str.rindex(sub), as_index(gs).str.rindex(sub))
+        assert_eq(
+            pd.Index(ps).str.rindex(sub),
+            as_index(gs).str.rindex(sub),
+            exact=False,
+        )
```

Comment:
What went on here?

Reply:
Yup, Index dtype became consistent in pandas 2.0. See above comment: https://github.com/rapidsai/cudf/pull/12962/files#r1153867852